src/main/java/edu/harvard/iq/dataverse/DvObject.java

```diff
@@ -493,6 +494,7 @@ public void setStorageQuota(StorageQuota storageQuota) {
      */
     public abstract boolean isAncestorOf( DvObject other );
     
+    public abstract PidProvider getEffectivePidGenerator();
```

Comment:
@qqmyers I've pulled this method up to the first common ancestor of the dataset and datafile. The only reason for this is that I wanted to reuse the `registerExternalIdentifier' method without much change.

Reply:
See other comments - I don't think files should have an effectivePidGenerator - you can always find the right one by looking up which provider can manage the file's globalId.

Reply:
Thanks, I'll revert those change and adapt my changed to obtain the PIDProvider via GlobalID.